client/plugins/components/vueFullscreen.ts

```diff
@@ -2,9 +2,28 @@ import Vue from 'vue';
 // @ts-expect-error - Missing type declerations
 import fullscreen from 'vue-fullscreen';
 
+type VueFullScreenOptions = {
+  callback?: (fullscreen: boolean) => void;
+  fullscreenClass?: string;
+  wrap?: boolean;
+  exitOnClickWrapper?: boolean;
+  background?: string;
+};
+interface VueFullscreen {
+  toggle(
+    target: Element,
+    options?: VueFullScreenOptions,
+    force?: boolean | undefined
+  ): void;
+  enter(target: Element, options?: VueFullScreenOptions): void;
+  exit(): void;
+  getState(): boolean;
+  support: boolean;
+}
+
```

Comment:
Why not add this upstream?

Reply:
@Camc314 Good question. Didn't bother because it could take more time and this has been sitting for a while already.

The problem I see with this is that the original project needs to be rewritten in TypeScript entirely, it's not a simple one-line change, and I'm not sure the original author wants that. I will see if I can propose the changes upstream but they might be rejected or abandoned because this.

Reply:
> The problem I see with this is that the original project needs to be rewritten in TypeScript entirely, it's not a simple one-line change, and I'm not sure the original author wants that. I will see if I can propose the changes upstream but they might be rejected or abandoned because this.

You can just add the typings to the upstream package without doing a rewrite, or submit them to DefinitelyTyped :smiley:

Reply:
@ferferga I have removed the types for now. They can be re-added when they're added either upstream (at vue-fullscreen) or at @types/vue-fullscreen